### PR TITLE
Fix canary not connecting to CW Agent

### PR DESCRIPTION
**Pull request by markkuhn**
*Description of changes:*
Canary is not connecting to the CW Agent.
This PR upgrades .NET from 3.1 to 6.0 and upgrades the CloudWatch Agent to the version 1.247354.0b251981.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: scripts/install-dotnet.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # print command before they are executed without expanding variables
 set -v
 
-# install the latest patch version of 3.1
-curl -sSL https://dot.net/v1/dotnet-install.sh | bash /dev/stdin -c 3.1
+# install the latest patch version of 6.0
+curl -sSL https://dot.net/v1/dotnet-install.sh | bash /dev/stdin -c 6.0
```
Comment (gordonpn) [inline scripts/install-dotnet.sh:6]: Why did the version need to be updated?

Comment (markkuhn) [reply scripts/install-dotnet.sh:6]: .NET 3.0 has reached end of life and is no longer supported. Changing to .NET 6.0 which is a supported release.

File: examples/Amazon.CloudWatch.EMF.Examples.Web/Properties/launchSettings.json [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
-      "launchBrowser": true,
-      "launchUrl": "weatherforecast",
-      "applicationUrl": "https://localhost:5001;http://localhost:5000",
-      "environmentVariables": {
-        "ASPNETCORE_ENVIRONMENT": "Development"
-      }
-    }
-  }
-}
+{
```
Comment (gordonpn) [inline examples/Amazon.CloudWatch.EMF.Examples.Web/Properties/launchSettings.json:1]: Weird, it marked the entire file as changed, but something seemed to have changed.

File: scripts/install-dotnet.sh (hunk 1 above)
Comment (gordonpn) [reply scripts/install-dotnet.sh:6]: `.NET Core 3.1` is indeed old, but technically still maintained.

If you've tested with `6.0` and nothing broke, then that's fine by me as well!

File: scripts/cwagent/Dockerfile [hunk 3]
```diff
@@ -4,7 +4,7 @@ RUN apt-get update &&  \
     apt-get install -y ca-certificates curl && \
     rm -rf /var/lib/apt/lists/*
 
-RUN curl -O https://amazon-cloud-watch-agent.s3.amazonaws.com/debian/amd64/1.237768.0/amazon-cloudwatch-agent.deb && \
+RUN curl -O https://s3.amazonaws.com/amazoncloudwatch-agent/ubuntu/amd64/latest/amazon-cloudwatch-agent.deb && \
```
Comment (gordonpn) [inline scripts/cwagent/Dockerfile:7]: Are we sure we want to use the latest version? Is it possible to pin a major version at least?

**Comment by gordonpn** [comment]
Upgrading the `.NET` version is what allowed CW Agent to be able to connect to the canary?

Or upgrading CW Agent did?

What version is CW Agent now?

File: scripts/install-dotnet.sh (hunk 1 above)
Comment (markkuhn) [reply scripts/install-dotnet.sh:6]: All tests pass with .NET 6. This change is not essential but end of life for .NET 3.1 is in 3 months.

File: scripts/cwagent/Dockerfile (hunk 3 above)
Comment (markkuhn) [reply scripts/cwagent/Dockerfile:7]: Good point. Will change that.

**Comment by markkuhn** [comment]
@gordonpn 

Screen Shot 2022-09-22 at 9 22 20 AM
These are the ECS logs before the change

Screen Shot 2022-09-22 at 9 23 17 AM
And this is after.

What fixed the issue was upgrading .NET to 6.0, upgrading the CodeBuild image to `amazonlinux2-x86_64-standard:4.0` and upgrading the CW Agent to latest.

File: examples/Amazon.CloudWatch.EMF.Examples.Web/Properties/launchSettings.json (hunk 2 above)
Comment (markkuhn) [reply examples/Amazon.CloudWatch.EMF.Examples.Web/Properties/launchSettings.json:1]: Yes, very odd.

**Review by gordonpn — APPROVED** [review]

**Review by gordonpn — APPROVED** [review]

**Review by Himtanaya — APPROVED** [review]

**Pull request merged**